pkg/arvo/app/btc-node-hook.hoon

```diff
@@ -0,0 +1,373 @@
+      =*  response  client-response.sign-arvo
+      =^  cards  state
+        ?+    +<.sign-arvo    (on-arvo:def wire sign-arvo)
+            %http-response    (http-response:bc wire response)
+        ==
+      [cards this]
+    ::
+    ++  on-fail   on-fail:def
+    --
+::
+|_  =bowl:gall
+::
+++  httr-to-rpc-response
+  |=  hit=httr:eyre
+  ^-  response:rpc:jstd
+  ~|  hit
+  :: ?.  ?=(%2 (div p.hit 100))
+  ::   fail+hit
+  =/  jon=json  (need (de-json:html q:(need r.hit)))
+  =,  dejs-soft:format
+  ^-  response:rpc:jstd
+  =;  dere
+    =+  res=((ar dere) jon)
+    ?~  res  (need (dere jon))
+    [%batch u.res]
+  |=  jon=json
+  ^-  (unit response:rpc:jstd)
+  =/  res=[id=(unit @t) res=(unit json) err=(unit json)]
+    %.  jon
+    =,  dejs:format
+    =-  (ou -)
+    :~  ['id' (uf ~ (mu so))]
+        ['result' (uf ~ (mu same))]
+        ['error' (uf ~ (mu same))]
+    ==
+  ?:  ?=([^ * ~] res)
+    `[%result [u.id.res ?~(res.res ~ u.res.res)]]
+  ~|  jon
+  ::%-  need
+  :: ((ot id+so error+(ot code+no message+so ~) ~) jon)
+  `[%error (handle-error jon)]
+::
+++  handle-error
+  |=  =json
+  ?~  json  ['' '' '']
+  %.  json
+  =,  dejs:format
+  =-  (ou -)
+  :~  =-  ['id' (uf '' (cu - (mu so)))]
+      |*(a=(unit) ?~(a '' u.a))
+      :-  'error'
+      =-  (uf ['' ''] -)
+      =-  (cu |*(a=(unit) ?~(a ['' ''] u.a)) (mu (ou -)))
+      :~  ['code' (uf '' no)]
+          ['message' (uf '' so)]
+  ==  ==
+::
+++  http-response
+  |=  [=wire response=client-response:iris]
+  ?.  ?=(%finished -.response)
+    [~ state]
+  =*  status    status-code.response-header.response
+  =/  rpc-resp=response:rpc:jstd
+    %-  httr-to-rpc-response
+    %+  to-httr:iris
+      response-header.response
+    full-file.response
+  ?.  ?=([%result *] rpc-resp)
+    ~&  [%error +.rpc-resp]
+    [~ state]
+  %-  handle-btc-response
+    (parse-response:btc-rpc:lib rpc-resp)
+::
+++  handle-btc-response
+  |=  btc-resp=btc-node-hook-response
+  ^-  (quip card _state)
+  :_  state
+  ^-  (list card)
+  ?+    -.btc-resp
+      ::  By default we just print all RPC responses that are not
+      ::  considered here explicitly for proper format printing or
+      ::  for being passed on to the store app.
+      ::
+      ~&  btc-resp
+      ~
+  ::
+      :: %abandon-transaction
+      :: %abort-rescan
+      :: %add-multisig-address
+      :: %backup-wallet
+      :: %bump-fee
+      %create-wallet
+    ^-  (list card)
+    =/  btc-store-req=btc-node-store-action
+      :+  %add-wallet   name.btc-resp
+      ?:(=('' warning.btc-resp) ~ (some warning.btc-resp))
+    [(btc-node-store-poke /store btc-store-req)]~
+  ::
+      :: %dump-privkey
+      :: %dump-wallet
+      :: %encrypt-wallet
+      :: %get-addresses-by-label
+  ::
+      %get-address-info
+    ^-  (list card)
+    ~&  [%address-info +.btc-resp]
+    ~
+  ::
+      %get-balance
+    ^-  (list card)
+    :_  ~
+    :*  %pass  /  %arvo  %d  %flog
+        %text  "amount={(trip +.btc-resp)}"
+    ==
```

Comment:
What's the reason for the fancy printing on this? Everything else just does `~&`. Maybe `+slog` is easier.

Reply:
This is a leftover of the `+ph` testing. I'll fix it